api/src/main/java/com/netflix/iceberg/encryption/EncryptionTypes.java

```diff
@@ -0,0 +1,50 @@
+package com.netflix.iceberg.encryption;
+
+import com.netflix.iceberg.types.Types;
+
+import java.nio.ByteBuffer;
+
+public class EncryptionTypes {
```

Comment:
This is quite a bit more complicated than what I envisioned. I assumed that this would be a single binary field, `key_ref`.

Thinking about this more, I can see how we will need more than a field in DataFile. That has to be passed into the FileIO for your use case, and maybe passed to file format implementations for using things like Parquet encryption. But I do want to make sure that what we add is minimal.

Reply:
With this much addition, I'm wondering if we should come up with a design doc for this?

Reply:
Yup I think the github ticket is becoming a little unwieldy for this scale of a design. I'd be ok with starting a design document for this.